### PR TITLE
Feature/evsemanager ocpp errorhandling

### PR DESCRIPTION
**Pull request by corneliusclaussen**
## Describe your changes

**Refactored Error Handling in EvseManager**
The following SessionEventEnums have been removed from the evse_manager interface:
* Error
* ErrorCleared
* AllErrorsCleared
* PermanentFault
* PermanentFaultCleared

The EvseManager now uses the error handling provided by the everest-framework to raise/report errors. A `PermanentFault` error raised by the EvseManager indicates that charging is not possible.

_____________________________

**Refactored Error Handling in OCPP and OCPP201 modules**

The OCPP modules where listening to the removed SessionEvents and were therefore refactored to be able to retrieve the errors raised by the new mechanism from the everest-framework.

Make use of everest-framework error handling feature in OCPP and OCPP201 modules:
* added enable_global_errors to OCPP and OCPP201 module to be able to subscribe to all errors reported in EVerest
* added error_mapping.hpp to OCPP module to be able to convert to MREC errors
* removed SessionEvent handling for ::Error and ::PermanentFault because this is replaced by subscribing to errors using the new tools of everest-framework

## Issue ticket number and link
Companion PR in libocpp: https://github.com/EVerest/libocpp/pull/673

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: modules/OCPP/OCPP.cpp [hunk 1]
```diff
@@ -369,6 +329,30 @@ void OCPP::init() {
...
+        } else {
+            this->event_queue[evse_id].push(error_info);
+        }
+    };
+
+    const auto error_cleared_handler = [this](const Everest::error::Error& error) {
+        const auto evse_id = error.origin.mapping.has_value() ? error.origin.mapping.value().evse : 0;
+
+        if (this->started) {
+            // TODO: Report correct evse_id once Error type includes it
```
Comment (Pietfried) [inline modules/OCPP/OCPP.cpp:347]: Remove

File: modules/OCPP/OCPP.cpp [hunk 2]
```diff
@@ -369,6 +329,30 @@ void OCPP::init() {
     invoke_init(*p_auth_provider);
     invoke_init(*p_data_transfer);
 
+    const auto error_handler = [this](const Everest::error::Error& error) {
+        const auto evse_id = error.origin.mapping.has_value() ? error.origin.mapping.value().evse : 0;
+        const auto error_info = get_error_info(error);
+        if (this->started) {
+            // TODO: Report correct evse_id once Error type includes it
```
Comment (Pietfried) [inline modules/OCPP/OCPP.cpp:336]: Remove

File: modules/OCPP/OCPP.cpp [hunk 3]
```diff
@@ -795,13 +779,30 @@ void OCPP::ready() {
...
-            while (!evse_session_event_queue.empty()) {
-                auto queued_session_event = evse_session_event_queue.front();
-                EVLOG_info << "Processing queued session event for evse_id: " << evse_id
-                           << ", event: " << queued_session_event.event;
-                this->process_session_event(evse_id, queued_session_event);
-                evse_session_event_queue.pop();
+        for (auto& [evse_id, evse_event_queue] : this->event_queue) {
+            // TODO: Probably we need a queue of std::variant here to be able to process raised errors as well
+            // Errors can only be processed after the start function, since only in there the ChargePointFSM is
+            // intialized
```
Comment (Pietfried) [inline modules/OCPP/OCPP.cpp:785]:
```suggestion
```

File: dependencies.yaml [hunk 4]
```diff
@@ -4,7 +4,7 @@
 ---
 everest-framework:
   git: https://github.com/EVerest/everest-framework.git
-  git_tag: v0.15.2
+  git_tag: feature/3-tier-model-support
```
Comment (Pietfried) [inline dependencies.yaml:7]:
```suggestion
  git_tag: v0.16.0
```

**Review by hikinggrass — CHANGES_REQUESTED** [review]

File: dependencies.yaml [hunk 5]
```diff
@@ -54,13 +54,13 @@ libcurl:
...
   git: https://github.com/EVerest/libevse-security.git
-  git_tag: b140c17b0a5eaf09b60035605ed8aeb84627eb78
+  git_tag: 8a300f39b94fb6364759dfbb57e8501facb94b2f
   cmake_condition: "EVEREST_DEPENDENCY_ENABLED_LIBEVSE_SECURITY"
 
 # OCPP
 libocpp:
   git: https://github.com/EVerest/libocpp.git
-  git_tag: 01f064f45f6d0ff44605db07373e4b9ea02b63b8
+  git_tag: feature/refactor-ocpp16-error-state-machine
```
Comment (Pietfried) [inline dependencies.yaml:63]:
```suggestion
  git_tag: 1067cf3ddf27be3a43f8abc519b69da11c4ef921
```

**Review by hikinggrass — APPROVED** [review]

**Pull request merged**